From d02032540bfe7959e018b7fe3602ce67eb30bc2a Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 30 Jan 2025 11:22:17 +0100
Subject: [PATCH 1502/1695] dts: st: define cryp2 node in stm32mp21xc.dtsi

Add node cryp2 in stm32mp21xc.dtsi.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e2ab0b93474dfb7c41b684f1ff8c2c2f6dfc9e94 dts: st: define cryp2 node in stm32mp21xc.dtsi]
Change-Id: I15c14029d355df34ab5cdad42f50f95a1658e2fe
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433034
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp21xc.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp21xc.dtsi b/core/arch/arm/dts/stm32mp21xc.dtsi
index 43ca3fccb..dc4267313 100644
--- a/core/arch/arm/dts/stm32mp21xc.dtsi
+++ b/core/arch/arm/dts/stm32mp21xc.dtsi
@@ -17,6 +17,16 @@
 		status = "disabled";
 	};
 
+	cryp2: cryp@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x400>;
+		interrupts = <GIC_SPI 157 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		access-controllers = <&rifsc STM32MP21_RIFSC_CRYP2_ID>;
+		status = "disabled";
+	};
+
 	saes: saes@42030c00 {
 		compatible = "st,stm32mp13-saes";
 		reg = <0x42030c00 0x400>;
-- 
2.47.2

